src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: string;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b <= c
+      || a + c <= b
+      || b + c <= a) {
+      throw new Error('Sides length should be correct');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
 
+    return Math.floor(Math.sqrt(p * (p - this.a)
+      * (p - this.b) * (p - this.c)) * 100) / 100;
+  }
 }
 
 export class Circle {
+  shape: string;
 
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.shape = 'circle';
+
+    if (radius <= 0) {
+      throw new Error('Radius length should be more than 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI
+      * this.radius
+      * this.radius
+      * 100)
+      / 100;
+  }
 }
 
 export class Rectangle {
+  shape: string;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+  ) {
+    this.shape = 'rectangle';
 
-export function getInfo(figure) {
+    if (a <= 0 || b <= 0) {
+      throw new Error('Sides length should be more than 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(this.a * this.b * 100) / 100;
```

Comment:
Don't repeat yourself: Math.floor( ... * 100) / 100 repeated 3 times in your code, you can create separate function for this operation